src/server/controllers/Grbl/GrblRunner.js

```diff
@@ -120,6 +120,20 @@ class GrblRunner extends events.EventEmitter {
             return;
         }
         if (type === GrblLineParserResultAlarm) {
+            const alarmPayload = {
+                activeState: GRBL_ACTIVE_STATE_ALARM
+            };
+            const nextState = {
+                ...this.state,
+                status: {
+                    ...this.state.status,
+                    ...alarmPayload
+                }
+            };
+            if (!_.isEqual(this.state.status, nextState.status)) {
+                this.state = nextState; // enforce change
+            }
+            this.emit('status', alarmPayload);
```

Comment:
Hi @Billiam

Have you ever tried removing the line:
`this.emit('status', alarmPayload);`

I think it will work as expected even if it does not emit the `status` event.

Reply:
The `status` event listener is used to control the status query and output Grbl status report to the UI console if a status query command (`?`) was requested.

However, I don't see the relevance of the `Alarm` event.

```js
this.runner.on('status', (res) => {
    this.actionMask.queryStatusReport = false;

    if (this.actionMask.replyStatusReport) {
        this.actionMask.replyStatusReport = false;
        this.emit('serialport:read', res.raw);
    }

    // Check if the receive buffer is available in the status report
    // @see https://github.com/cncjs/cncjs/issues/115
    // @see https://github.com/cncjs/cncjs/issues/133
    const rx = Number(_.get(res, 'buf.rx', 0)) || 0;
    if (rx > 0) {
        // Do not modify the buffer size when running a G-code program
        if (this.workflow.state !== WORKFLOW_STATE_IDLE) {
            return;
        }

        // Check if the streaming protocol is character-counting streaming protocol
        if (this.sender.sp.type !== SP_TYPE_CHAR_COUNTING) {
            return;
        }

        // Check if the queue is empty
        if (this.sender.sp.dataLength !== 0) {
            return;
        }

        // Deduct the receive buffer length to prevent from buffer overrun
        const bufferSize = (rx - 8); // TODO
        if (bufferSize > this.sender.sp.bufferSize) {
            this.sender.sp.bufferSize = bufferSize;
        }
    }
});
```

Reply:
@cheton  Thanks. I don't need that emit for this bug, I just misunderstood it's purpose (and thought it was desired here also). I'll update the PR to remove it and update the test.

Reply:
Removed, and updated the test to check the runner state in the alarm event.